### PR TITLE
Ensures text manager is not loaded before the shell is initialized.

### PR DESCRIPTION
**Pull request by zooba**
Ensures text manager is not loaded before the shell is initialized. (This helps prevent the Tab setting from being incorrect)
Switches to correct parsing of Unicode identifiers.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Extensions.cs [hunk 1]
```diff
@@ -826,6 +827,51 @@ public static class Extensions {
...
+                }
+                return VSConstants.S_OK;
+            }
+        }
+
+        internal static Task WaitForShellInitializedAsync(this IServiceProvider provider) {
+            if (provider.IsShellInitialized()) {
+                return Task.FromResult<object>(null);
+            }
+            return new ShellInitializedNotification(provider.GetShell()).Task;
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/Extensions.cs]: Is there a risk of waiting forever here if the shell initializes after you've checked, but before you register for the property changes event?

Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Extensions.cs]: Yes. We should check once more before returning

File: Python/Product/PythonTools/PythonToolsService.cs [hunk 2]
```diff
@@ -579,13 +561,16 @@ public sealed class PythonToolsService : IDisposable {
 
         internal System.IServiceProvider Site => _container;
 
-        internal LanguagePreferences LangPrefs {
-            get {
-                return _langPrefs;
+        internal LanguagePreferences LangPrefs => _langPrefs.Value;
+
+        internal async Task<LanguagePreferences> GetLangPrefsAsync() {
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonToolsService.cs]: I think that deserves a comment.  If I need preferences, how do I know if I'm supposed to call the method or the propery?

**Comment by huguesv** [comment]
:+1:

**Pull request merged**